cmd/root.go

```diff
@@ -102,16 +103,36 @@ of Cloud Storage FUSE, see https://cloud.google.com/storage/docs/gcs-fuse.`,
 // into POSIX compliant args. All it does is that it converts flags specified
 // using a single-hyphen to double-hyphens. We are excluding "-v" because it's
 // reserved for showing version in Cobra.
-func ConvertToPosixArgs(args []string) []string {
+func ConvertToPosixArgs(args []string, c *cobra.Command) []string {
 	pArgs := make([]string, 0, len(args))
+	flagSet := make(map[string]bool)
+	c.PersistentFlags().VisitAll(func(f *pflag.Flag) {
+		flagSet[f.Name] = true
+	})
 	for _, a := range args {
 		switch {
 		case a == "--v", a == "-v":
 			pArgs = append(pArgs, "-v")
 		case a == "--h", a == "-h":
 			pArgs = append(pArgs, "-h")
-		case strings.HasPrefix(a, "-") && !cfg.IsNegativeNumber(a) && !strings.HasPrefix(a, "--"):
-			pArgs = append(pArgs, "-"+a)
+		case strings.HasPrefix(a, "-"):
+			// Remove the string post the "=" sign.
+			// This converts -a=b to -a but an arg like --c would remain unchanged.
+			flg, _, _ := strings.Cut(a, "=")
+			// Remove one hyphen from the beginning.
+			// This converts --a -> -a and -a -> a.
+			// If no hyphen is found then leave it as is and found is set to false.
+			flg, found := strings.CutPrefix(flg, "-")
+
+			if found && flagSet[flg] {
+				// "a" is a full-form flag which has been specified with a single hyphen.
+				// So add another hyphen so that pflag processes it correctly.
+				pArgs = append(pArgs, "-"+a)
+			} else {
```

Comment:
We can't remove the edge block since it won't handle the cases where the flag argument itself has double-hyphens such as `gcsfuse --app-name '--abc' <bkt> <mount_point>`. While we are still not handling the cases where someone passes the flag name itself as value e.g.`gcsfuse --app-name '--app-name' <bkt> <mount_point>` but it sure helps reduce the surface area.

Reply:
We can capture flag name using regexp as well: 
```
re := regexp.MustCompile(`-([^=]+)(=(.*))?`)
matches := re.FindStringSubmatch(a)
flg := matches[1]
```

Also, shall we update this comment: 

// Either "a" is not a flag or "a" has been specified with two-hyphens already. 
to 
// "a" is flag value so append it as it is.

Reply:
Thanks. Updated the comment.
I'd avoid the regex since it's easy to get wrong and a bit unreadable.